### ISSUE TITLE
Python Illegal wire type for field Message.Field .PeopleRequest.ages: 2 (0 expected))

### ISSUE THREAD

**Issue by sulliwane** [grpc/grpc#9439]
Hello,
here is my proto:
```proto
message PeopleRequest {
  repeated string names = 1; // ex: ['jack', 'marie']
  repeated int32 ages = 2; // ex: [1, 2]
}
```
and here is my call:
```python
people_request = grpc.PeopleRequest(
      names=['daniel'],
      ages=[32],
)
print(people_request) // Ok, no error when building the people_request object
stub.GetPeoples(people_request) // ERROR Illegal wire type for field ages: 2 (0 expected))
```
This simple grpc call throws an error about the type of the field **ages**. I really can't see where is the problem...That's a simple list of integers isn't it? Why it seems to recognize it as a list of strings/bytes (wire type 2) ?

Many thanks for you help!

**Comment by sulliwane** [comment]
And here is the auto-generated python code:
```python
_descriptor.FieldDescriptor(
      name='names', full_name='PeopleRequest.names', index=0,
      number=1, type=9, cpp_type=9, label=3,
      has_default_value=False, default_value=[],
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      options=None),
_descriptor.FieldDescriptor(
      name='ages', full_name='PeopleRequest.ages', index=1,
      number=2, type=5, cpp_type=1, label=3,
      has_default_value=False, default_value=[],
      message_type=None, enum_type=None, containing_type=None,
      is_extension=False, extension_scope=None,
      options=None),
```

**Comment by nathanielmanistaatgoogle** [comment]
I don't think that `print(people_request) // Ok, no error when building the people_request object` is sufficient to demonstrate that your `people_request` message is legitimately constructed. You write that "the error is [...] not when building the protobuf `people_request` object", but for performance reasons Python Protocol Buffers don't do any validation on construction, only when they are serialized. Try something like
```python
try:
  serialized_people_request = grpc.PeopleRequest.SerializeToString(people_request)
except Exception as e:
  print(e)
else:
  print('Serialized to binary of type {} and length {}'.format(type(serialized_people_request), len(serialized_people_request)))
```
? I suspect that the issue you're having is more on the Protocol Buffers side of the situation than the gRPC side.

**Comment by sulliwane** [comment]
Ok, I didn't know there was no check when building the object. So indeed my reasoning isn't correct.

When serializing this object:
```python
{ 'names': ['daniel', 'jack'], 'ages': [18, 22] }
```
here is the console output of your test code:
```bash
Serialized to binary of type <class 'bytes'> and length 20
```
After what, the same error is thrown when calling the grpc method:
```bash
grpc._channel._Rendezvous: <_Rendezvous of RPC that terminated with (StatusCode.INTERNAL, Illegal wire type for field Message.Field grpc.GetPeoples.ages: 2 (0 expected))>
```

Should I post this issue on the protocol buffers github's page?

Many thanks for your help!

**Comment by nathanielmanistaatgoogle** [comment]
I'd like to see more of your code. Where is(are) the `Service` portion(s) of your `.proto` file? How are you constructing your `stub`?

**Comment by sulliwane** [comment]
Of course, here is the whole code:
```proto
syntax = "proto3";

package grpcClient;

service GrpcClient {
  rpc GetPeople(PeopleRequest) returns (PeopleResponse) {}
}
message PeopleRequest {
  repeated string names = 1; // ex: ['jack', 'marie']
  repeated int32 ages = 2; // ex: [18, 24]
}
message PeopleResponse {
  repeated Person people = 1;
}
message Person {
  string name = 1;
  int32 age = 2;
}
```
Stub construction:
```python
from . import grpcClient_pb2
class GrpcClient:
    def __init__(self):
        server_url = "http://my_url:5001"
        secure_channel = make_secure_channel(server_url)
        self.__stub = grpcClient_pb2.GrpcClientStub(secure_channel)

    def get_people(self, **kwargs):
        people_request = grpcClient_pb2.PeopleRequest(**kwargs)
        # your test example
        try:
            serialized_people_request = grpcClient_pb2.PeopleRequest.SerializeToString(ins)
        except Exception as e:
            print(e)
        else:
            print('Serialized to binary of type {} and length {}'.format(type(serialized_people_request), len(serialized_people_request)))
        return self.__stub.GetPeople(people_request) # this throw an error
```
The main logic:
```python
grpcClient = GrpcClient()
grpcClient.get_people({ 'names': ['daniel', 'jack'] }) # this is OK
grpcClient.get_people({ 'names': ['daniel', 'jack'], 'ages': [18, 22] }) # this throws
```

**Comment by kpayson64** [comment]
I don't believe protobuf supports direct assignment of repeated fields in the constructor.
See https://developers.google.com/protocol-buffers/docs/reference/python-generated#repeated-fields .

**Comment by sulliwane** [comment]
@kpayson64 appending to the list instead of assigning throws the same error:
```python
from . import grpcClient_pb2

def run():
    server_url = "http://my_url:50051"
    secure_channel = make_secure_channel(server_url)
    stub = grpcClient_pb2.GrpcClientStub(secure_channel)
    people_request = grpcClient_pb2.PeopleRequest()
    people_request.ranks.append(1)  # appending here as demonstrated in the guide https://developers.google.com/protocol-buffers/docs/reference/python-generated#repeated-fields
    people_response = grpcClient_pb2.GetPeople(people_request) #  throws grpc._channel._Rendezvous: <_Rendezvous of RPC that terminated with (StatusCode.INTERNAL, Illegal wire type for field Message.Field .grpcClient_pb2.PeopleRequest.ranks: 2 (0 expected))>

if __name__ == '__main__':
    run()
```

@nathanielmanistaatgoogle Out of the two .py generated (**grpcClient_pb2** and **grpcClient_pb2_grpc**), I just use one: **grpcClient_pb2** to generate the stub, and build the objects. The other file ending with **_grpc** -> **grpcClient_pb2_grpc** seems broken, with many duplicated imports at the beginning of the file:
```python
import grpc
from grpc.framework.common import cardinality
from grpc.framework.interfaces.face import utilities as face_utilities

import grpcClient_pb2 as grpcClient__pb2
import grpcClient_pb2 as grpcClient__pb2
import grpcClient_pb2 as grpcClient__pb2
import grpcClient_pb2 as grpcClient__pb2
import grpcClient_pb2 as grpcClient__pb2
import grpcClient_pb2 as grpcClient__pb2
import grpcClient_pb2 as grpcClient__pb2
import grpcClient_pb2 as grpcClient__pb2
```
So this could be related to the versions I'm using (python 3.6/grpcio  1.0.4/protobuf 3.1.0.post1)...or some mistake when generating the .py files....or I don't know. I will try with python 3.5.

**Comment by sulliwane** [comment]
just resintalled all the packages with python 3.5, same error :-(

**Comment by sulliwane** [comment]
**Short story:** My server was written in node.js, and:
```
rm -rf node_modules
npm i
```
solved the problem for me.

**Long story:** It appears that whatever the server my client was connecting to (I had a local and a remote node.js server running for the test) in both case my client would throw. That's why I looked for an error in the client code, and not on the server side.

Finally I reimplemented the server side in python, and surprisingly my client wouldn't throw when connecting to the python server. So I started to suspect my node.js server had a problem, I finally completely reinstalled the **node_modules** of my node.js server, restarted, and indeed the problem disappeared. So I conclude the problem was related to some package in **node_modules**.
